trajectory_tracker/src/trajectory_recorder.cpp

```diff
@@ -66,74 +65,72 @@ class RecorderNode
   double ang_interval_;
   bool store_time_;
 
-  ros::NodeHandle nh_;
-  ros::NodeHandle pnh_;
-  ros::Publisher pub_path_;
-  tf2_ros::Buffer tfbuf_;
-  tf2_ros::TransformListener tfl_;
-  ros::ServiceServer srs_clear_path_;
+  rclcpp::Publisher<nav_msgs::msg::Path>::SharedPtr pub_path_;
+  std::unique_ptr<tf2_ros::Buffer> tfbuf_;
+  std::shared_ptr<tf2_ros::TransformListener> tfl_;
+  rclcpp::Service<std_srvs::srv::Empty>::SharedPtr srs_clear_path_;
 
-  nav_msgs::Path path_;
+  nav_msgs::msg::Path path_;
 };
 
 RecorderNode::RecorderNode()
-  : nh_()
-  , pnh_("~")
-  , tfl_(tfbuf_)
+  : rclcpp::Node("trajectory_recorder")
 {
-  neonavigation_common::compat::checkCompatMode();
-  pnh_.param("frame_robot", frame_robot_, std::string("base_link"));
-  pnh_.param("frame_global", frame_global_, std::string("map"));
-  neonavigation_common::compat::deprecatedParam(pnh_, "path", topic_path_, std::string("recpath"));
-  pnh_.param("dist_interval", dist_interval_, 0.3);
-  pnh_.param("ang_interval", ang_interval_, 1.0);
-  pnh_.param("store_time", store_time_, false);
-
-  pub_path_ = neonavigation_common::compat::advertise<nav_msgs::Path>(
-      nh_, "path",
-      pnh_, topic_path_, 10, true);
-  srs_clear_path_ = pnh_.advertiseService("clear_path", &RecorderNode::clearPath, this);
+  frame_robot_ = this->declare_parameter<std::string>("frame_robot", "base_link");
+  frame_global_ = this->declare_parameter<std::string>("frame_global", "map");
+  topic_path_ = this->declare_parameter<std::string>("path", "recpath");
+
+  dist_interval_ = this->declare_parameter<double>("dist_interval", 0.3);
+  ang_interval_ = this->declare_parameter<double>("ang_interval", 1.0);
+  store_time_ = this->declare_parameter<bool>("store_time", false);
+
+  pub_path_ = this->create_publisher<nav_msgs::msg::Path>("path", rclcpp::QoS(10).transient_local());
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+  srs_clear_path_ = this->create_service<std_srvs::srv::Empty>(
+    "clear_path", std::bind(&RecorderNode::clearPath, this, _1, _2));
```

Comment:
`srs_clear_path_` should have namespace "~"
```suggestion
    "~/clear_path", std::bind(&RecorderNode::clearPath, this, _1, _2));
```